service/history/shard/context_impl.go

```diff
@@ -1889,9 +1890,6 @@ func (s *ContextImpl) acquireShard() {
 	// lifecycleCtx. The persistence operations called here use lifecycleCtx as their context,
 	// so if we were blocked in any of them, they should return immediately with a context
 	// canceled error.
-	policy := backoff.NewExponentialRetryPolicy(1 * time.Second).
```

Comment:
This moves the constants pretty far from the comment that describe what is going on. What do you think about having this do
```
policy := s.acquireShardRetryPolicy
if policy == nil {
  policy = backoff.NewExponentialRetryPolicy(...)
}
```
and then only setting an override in the test? It's a little messy also.

Alternatives: put the policy in a top-level const or var and add a little comment, or make the 5m a dynamic config (arguably it should have been)

Reply:
I think that's great. Fixed

Reply:
I was suggesting making the real policy the default, not the test policy. The point is that the comment right above here talks about timing (i.e. "5 minutes"), so the code for that policy should also be right here. Otherwise the actual code (`WithExpirationInterval(5*time.Minute)`) appears very far from any context and is hard for a reader to understand.

I'd also suggest not passing in the policy in the constructor but just having tests do `s.acquireShardRetryPolicy = ...` so that the override is right next to the test code, which makes the tests easier to read too (the reader doesn't have to wonder why you did `.Times(6)` or whether it's possible the retries will exceed 1s).

The non-test code can just let it default to nil when constructing and then get the actual policy here.

Reply:
Oops. I mixed these up. Fixed